packages/odyssey-react-mui/src/RadioGroup.tsx

```diff
@@ -73,31 +81,64 @@ const RadioGroup = ({
   HintLinkComponent,
   id: idOverride,
   isDisabled,
+  isReadOnly = false,
   label,
   name: nameOverride,
   onChange: onChangeProp,
   testId,
   translate,
   value,
 }: RadioGroupProps) => {
+  const [selectedValue, setSelectedValue] = useState(value ?? defaultValue);
+
   const controlledStateRef = useRef(
     getControlState({
       controlledValue: value,
       uncontrolledValue: defaultValue,
     }),
   );
+
   const inputValues = useInputValues({
     defaultValue,
     value,
     controlState: controlledStateRef.current,
   });
 
   const onChange = useCallback<NonNullable<MuiRadioGroupProps["onChange"]>>(
-    (event, value) => {
-      onChangeProp?.(event, value);
+    (event, newValue) => {
+      if (!isReadOnly && !isDisabled) {
+        setSelectedValue(newValue); // Update internal state
+        onChangeProp?.(event, newValue); // Notify parent
+      }
+    },
+    [onChangeProp, isReadOnly, isDisabled],
+  );
+
+  const handleMouseDown = useCallback(
```

Comment:
I think this needs to be handled in an `onClick` or `onChange`. My suspicion is that a user can still update this value via keyboard in the `readonly` state

Reply:
Good call, mouseDown triggers before onClick so I was using that after having problems with onChange allowing the value to be changed in `readonly`, but it looks like onClick works, and is more conventional, plus supports keyboard and mouse.